### PR TITLE
fix: added container wrapper component to handle styles

### LINKED ISSUE #2587: Container styles
Containers should have more styles than the current rounded borders and shadow

### LINKED ISSUE #4419: [Feature] Text widget should have padding controls
## Summary
The text widget has no padding controls so it looks ugly when you color the background.

Screen Shot 2021-05-11 at 4 32 32 AM
Providing padding controls will allow the developer to create better looking text cards.

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> Added container wrapper for styling

Fixes #2587 
Fixes #4419 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Tested locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: FIX/2587-container-style-support 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.45 **(0.02)** | 36.27 **(-0.01)** | 33.59 **(0.03)** | 55 **(0.01)**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/WidgetStyleContainer.tsx** | **80.77** | **37.5** | **100** | **80.77**
 :red_circle: | app/client/src/widgets/ContainerWidget/component/index.tsx | 93.33 **(0.23)** | 63.16 **(-1.84)** | 100 **(0)** | 93.33 **(0.23)**
 :red_circle: | app/client/src/widgets/FormWidget/widget/index.tsx | 86.96 **(-0.8)** | 52.63 **(0)** | 91.67 **(-0.64)** | 88.1 **(-0.79)**</details>

### CODE REVIEW

**Review by marks0351 — CHANGES_REQUESTED** [review]
Great job on adding the style features!!
Pls add a docs update PR for these changes.
when border is more than a certain value the widget overflows widget boundaries. it should not be the case.

**Comment by vibhandikyash** [comment]
@marks0351 Here also updated Border styles for Text Widget for issue #4419

**Comment by marks0351** [comment]
@vibhandikyash when a container radius is set very large the container goes away which is fine and is expected but the area below container becomes unaccessible because of the div overflow. 
Also when border radius is set to a large number it does not work as expected with text widget. text widget starts to overflow.

**Review by marks0351 — CHANGES_REQUESTED** [review]
@vibhandikyash when a container radius is set very large the container goes away which is fine and is expected but the area below container becomes unaccessible because of the div overflow. 
Also when border radius is set to a large number it does not work as expected with text widget. text widget starts to overflow.

**Review by somangshu — DISMISSED** [review]
Approving to run tests

**Comment by somangshu** [comment]
@vibhandikyash related test has failed, Please check

**Review by marks0351 — APPROVED** [review]

**Comment by somangshu** [comment]
Rerunning all jobs here, `Draggable_PropertyPane_Widgets_spec.js` failed because the property pane moved while widget is dragged down. Not related to the PR

**Comment by somangshu** [comment]
Failed again, @vibhandikyash please check this locally and let me know

**Review by somangshu — DISMISSED** [review]
LGTM, @rishabhsaxena thanks for the help here. 

Approving to run the tests

**Comment by somangshu** [comment]
@marks0351 @vibhandikyash here is my feedback

### Container widget:
1. The old property `background-colour` should be above the new styles
2. All the properties should be under the style section
3. Same for for the form widget
4. The border is always cut from the side where there is a border radius Screenshot 2021-09-03 at 12 36 37 PM
5. For older container / creating apps from DB the new properties are undefined which other wise is 0 when initially dragged and dropped


### Text widget:
1. OP requested for the margin property, which is not implemented (is this intended)
2. Other properties were not requested in the text widget, today placing the text widget in the container widget get you the attributes you need. So in my view, these new properties are not required in the text widget

**Comment by YogeshJayaseelan** [comment]
@vibhandikyash Currently all the border style related properties comes under general section and it is be separated and placed under style property 

Screenshot 2021-09-03 at 3 38 57 PM

**Comment by marks0351** [comment]
@vibhandikyash the text widget and container still overflows when border width given is high.

**Review by marks0351 — CHANGES_REQUESTED** [review]
@vibhandikyash  the text widget and container still overflows when border width given is high.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/PropertyPane/Draggable_PropertyPane_Widgets_spec.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ describe("Widget property pane draggable", function() {
       cy.openPropertyPane(testWidget);
       cy.get("#popper-draghandler")
         .trigger("mousedown", { which: 1 })
-        .trigger("mousemove", { clientX: 400, clientY: 500 })
```
Comment (marks0351) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/PropertyPane/Draggable_PropertyPane_Widgets_spec.js:52]: @NandanAnantharamu can you pls review/validate this change?

**Review by marks0351 — DISMISSED** [review]

**Review by NandanAnantharamu — DISMISSED** [review]
LGTM

**Review by marks0351 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@vibhandikyash Verified the container style functionality and looks good to me

**Comment by vivekverma2312** [comment]
@vibhandikyash Working fine for text widget too.

**Pull request merged**